### PR TITLE
Update .editorconfig

### PR DESCRIPTION
**Pull request by vinkla**
Added better support for `*.md` and `*.yml` files.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 indent_style = space
 indent_size = 4
 trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.yml]
+indent_style = space
+indent_size = 2
```
Comment (keradus) [inline .editorconfig]: Please use 4 instead

**Comment by vinkla** [comment]
4 spaces in Yaml? Isn't 2 required?

**Comment by SpacePossum** [comment]
>4 spaces in Yaml? Isn't 2 required?

it is not required, please see http://www.yaml.org/spec/current.html#id2519916

**Comment by vinkla** [comment]
Done.

**Comment by GrahamCampbell** [comment]
:+1:

File: .editorconfig [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 [*]
+charset = utf-8
 end_of_line = lf
 insert_final_newline = true
 indent_style = space
 indent_size = 4
 trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (vinkla) [inline .editorconfig]: `CHANGELOG.md`, `CONTRIBUTING.md` and `UPGRADE.md`.

Comment (keradus) [inline .editorconfig]: why do we want to have trailing whitespaces at those files ?

Comment (vinkla) [inline .editorconfig]: In Markdown, two spaces in the end of a line means line-break.

https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#lines

Comment (keradus) [inline .editorconfig]: we don't rely on that feature. i would not allow for trailing spaces anywhere except fixture files

Comment (vinkla) [inline .editorconfig]: Which means you don't want this?

Comment (keradus) [inline .editorconfig]: indeed, i would drop special rule for .md

**Comment by vinkla** [comment]
I'll just go ahead on close this all together.

**Pull request closed without merging**